File: solutions/102.binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> res;
        helper(res, root, 0); 
        return res;
    }
private:
    void helper(vector<vector<int>>& res, TreeNode* root, int level) {
        if(!root) return;
        if(res.size() == level) 
            res.push_back(vector<int>());
        res[level].push_back(root->val);
        helper(res, root->left, level + 1);
        helper(res, root->right, level + 1);
    }
};

File: solutions/102.binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        if(root == null) return res;
        List<Integer> curRes = null;
        Queue<TreeNode> queue = new LinkedList<>();
        TreeNode curNode = null;
        queue.offer(root);
        while(!queue.isEmpty()) {
            curRes = new ArrayList<>();
            for(int size = queue.size(); size > 0; size--) {
                curNode = queue.remove();
                curRes.add(curNode.val);
                if(curNode.left != null) queue.offer(curNode.left);
                if(curNode.right != null) queue.offer(curNode.right);
            }
            res.add(curRes);
        }
        return res;
    }
}